### ISSUE TITLE
Inference mode dupe specializes without guards

### ISSUE THREAD

**Issue by ezyang** [pytorch/torchdynamo#1939]
### 🐛 Describe the bug

```
import torch
import torch._dynamo

@torch._dynamo.optimize("aot_eager")
def f(x, y):
    return (x + y) * 1

x = torch.tensor([1.])
y = torch.tensor([2.])
print(f(x, x))
print(f(x, y))
```

Adding guards could help, but probably better to just detach all the inputs before tracing.
